tutorials/text/deduplication/semantic/semantic_step_by_step.ipynb

```diff
@@ -0,0 +1,1114 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b47ce381",
+   "metadata": {},
+   "source": [
+    "#### Looking at duplicates\n",
+    "\n",
+    "- `id` : This is a list of all ids that are above our similarity threshold `eps`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "865a9273",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>id</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>656</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>711</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>807</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>842</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>868</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "    id\n",
+       "0  656\n",
+       "1  711\n",
+       "2  807\n",
+       "3  842\n",
+       "4  868"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pd.read_parquet(os.path.join(duplicates_output_path, os.listdir(duplicates_output_path)[0])).head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88140f70",
+   "metadata": {},
+   "source": [
+    "## Removing Duplicates\n",
+    "\n",
+    "1. We load the persisted id generator\n",
+    "2. Run the removal workflow\n",
+    "3. Kill the id generator\n",
+    "\n",
+    "### Notes\n",
+    "1. When running removal workflow, we must specify the same input configuration as we did when we \"generated ids\".\n",
+    "2. In this tutorial that happened at the embedding generation step.\n",
+    "3. Therefore its required that we match the saame arguments of filepath, filetype and files_per_partition / blocksize.\n",
+    "4. This is required because ids are generated based on hash(filenames) in each task. If the same hash is not found in the id generator it'll error out.\n",
+    "\n",
+    "### Performance\n",
+    "1. If you notice OOMs during this stage you can try using RayDataActor\n",
```

Comment:
Will this be in the docs somewhere? Otherwise I think this instruction is a bit confusing to users first becoming acquainted with Curator.